src/bridge/SequencerInbox.sol

```diff
@@ -260,16 +389,81 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
                 prevMessageCount_,
                 newMessageCount_
             );
-        if (seqMessageIndex != sequenceNumber_ && sequenceNumber_ != ~uint256(0))
+
+        // ~uint256(0) is type(uint256).max, but ever so slightly cheaper
+        if (seqMessageIndex != sequenceNumber_ && sequenceNumber_ != ~uint256(0)) {
             revert BadSequencerNumber(seqMessageIndex, sequenceNumber_);
+        }
+
         emit SequencerBatchDelivered(
             seqMessageIndex,
             beforeAcc,
             afterAcc,
             delayedAcc,
             totalDelayedMessagesRead,
             timeBounds_,
-            BatchDataLocation.TxInput
+            IBridge.BatchDataLocation.TxInput
+        );
+    }
+
+    function addSequencerL2BatchFromBlob(
```

Comment:
nit: I think this should be "FromBlobs" plural since it can use multiple blobs

Reply:
Thanks! merged